src/pages/index.js

```diff
@@ -143,77 +139,121 @@ const editProfilePopup = new PopupWithForm(
   profileDescriptionInput.value = userData.description;
   formValidators["profile-form"].resetValidation();
-  /* openModal(profileEditModal); */
   editProfilePopup.open();
 });
 
 addButton.addEventListener("click", () => {
-  // openModal(addCardModal);
   addCardPopup.open();
 });
 
-// Save Button Event Listeners
-// addCardForm.addEventListener("submit", handleAddCardFormSubmit);
-// profileEditForm.addEventListener("submit", handleProfileFormSubmit);
-
-// Combining overlay and close button listeners
-// const popups = document.querySelectorAll(".modal");
-
-// popups.forEach((popup) => {
-//   popup.addEventListener("mousedown", (evt) => {
-//     if (
-//       evt.target.classList.contains("modal_opened") ||
-//       evt.target.classList.contains("modal__close")
-//     ) {
-//       // closeModal(popup);
-//       popup.classList.remove("modal_opened");
-//     }
-//   });
-// });
-
 // ---------------- Initialize Form Validators ----------------
-
-// const settings = {
-//   inputSelector: ".modal__input",
-//   submitButtonSelector: ".modal__button",
-//   inactiveButtonClass: "modal__button_inactive",
-//   inputErrorClass: "modal__input_type_error",
-//   errorClass: "modal__error_visible",
-//   formSelector: ".modal__form",
-// };
-
-// Define an object for storing validators
 const formValidators = {};
 
 const enableValidation = (config) => {
   const formList = Array.from(document.querySelectorAll(config.formSelector));
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
-    // Get the name of the form
     const formName = formElement.getAttribute("name");
-
-    // Store the validator using the `name` of the form
     formValidators[formName] = validator;
     validator.enableValidation();
   });
 };
 
 enableValidation(settings);
+
+// ---------------- Delete Popup ----------------
+const deletePopup = document.getElementById("delete-popup");
+const confirmDeleteButton = document.getElementById("confirm-delete");
+
+function handleLikeCard(cardId) {
+  api.deleteACard(cardId).then((response) => {
+    console.log(response);
+    return response;
+  });
+  // .catch(error => console.error('Error:', error));
+}
+
+// Function to handle like button click
+function handleLikeButtonClick(card) {
+  console.log(card);
+  if (card.isLiked) {
+    api
+      .dislikeacard(card._data._id)
+      .then((response) => {
+        console.log(response);
+        card.setLikeStatus(response.isLiked);
+      })
+      .catch((error) => console.error("Error:", error));
+  } else {
+    api
+      .likeaCard(card._data._id)
+      .then((response) => {
+        console.log(response);
+        card.setLikeStatus(response.isLiked);
+      })
+      .catch((error) => console.error("Error:", error));
+  }
+}
+
+// Add event listener to like buttons
+document.querySelectorAll(".like-button").forEach((button) => {
+  button.addEventListener("click", handleLikeButtonClick);
+});
```

Comment:
* Please, clean up `index.js` from invalid code of like handling and form submissions. You should do that via your classes

Reply:
Now only the `Card` class knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle